src/components/Griff/index.tsx

```diff
@@ -0,0 +1,444 @@
+        drawLines,
+        drawPoints,
+        loader,
+        pointWidth,
+        strokeWidth,
+        opacity,
+        opacityAccessor,
+        pointWidthAccessor,
+        timeAccessor,
+        xAccessor,
+        x0Accessor,
+        x1Accessor,
+        yAccessor,
+        y0Accessor,
+        y1Accessor,
+        timeDomain,
+        timeSubDomain,
+        xDomain,
+        xSubDomain,
+        yDomain,
+        ySubDomain,
+        updateInterval,
+      };
+      const collection: IncomingCollection =
+        series.collectionId !== undefined
+          ? collectionsById[series.collectionId] || { id: '' }
+          : { id: '' };
+      const completedSeries: BaseSeries = combineItems(
+        DEFAULT_SERIES_CONFIG,
+        dataProvider,
+        collection,
+        series
+      );
+      completedSeries.timeDomain = copyDomain(
+        completedSeries.timeDomain,
+        DomainPriority.GRIFF
+      );
+      return [...acc, completedSeries];
+    }, []);
+  };
+
+  registerCollection = ({ id, ...collection }: IncomingCollection) => {
+    this.setState(({ collectionsById }) => ({
+      collectionsById: {
+        ...collectionsById,
+        [id]: deleteUndefinedFromItem({
+          ...collection,
+          id,
+        }),
+      },
+    }));
+
+    // Return an unregistration so that we can do some cleanup.
+    return () => {
+      this.setState(({ collectionsById }) => {
+        const copy = { ...collectionsById };
+        delete copy[id];
+        return {
+          collectionsById: copy,
+        };
+      });
+    };
+  };
+
+  updateCollection = ({ id, ...collection }: IncomingCollection) => {
+    this.setState(({ collectionsById }) => ({
+      collectionsById: {
+        ...collectionsById,
+        [id]: deleteUndefinedFromItem({
+          ...collectionsById[id],
+          ...collection,
+          id,
+        }),
+      },
+    }));
+  };
+
+  registerSeries = ({ id, ...series }: IncomingSeries) => {
+    this.setState(({ seriesById }) => ({
+      seriesById: {
+        ...seriesById,
+        [id]: deleteUndefinedFromItem({
+          ...series,
+          id,
+        }),
+      },
+    }));
+
+    // Return an unregistration so that we can do some cleanup.
+    return () => {
+      this.setState(({ seriesById }) => {
+        const copy = { ...seriesById };
+        delete copy[id];
+        return {
+          seriesById: copy,
+        };
+      });
+    };
+  };
+
+  updateSeries = ({ id, ...series }: IncomingSeries) => {
+    this.setState(({ seriesById }) => ({
+      seriesById: {
+        ...seriesById,
+        [id]: deleteUndefinedFromItem({
+          ...seriesById[id],
+          ...series,
+          id,
+        }),
+      },
+    }));
+  };
+
+  // Add a helper method to render the legacy props using the new tree structure
+  // format. This is only intended to ease the transition pain and is not
+  // intended to be an ongoing solution.
+  renderLegacyItems = () => {
```

Comment:
Is this guy obsolete now?

Reply:
Yeah I think we can remove this since we're making a huge breaking API change anyway.